File: test_latency.js
const MongoClient = require('mongodb').MongoClient;
const assert = require('assert');

// Connection URL
const url = 'mongodb+srv://veronica:<PASSWORD>@pluto.e3h8i.mongodb.net/test?retryWrites=true&w=majority';

// Database Name
const dbName = 'test';

// Create a new MongoClient
const client = new MongoClient(url);

// Use connect method to connect to the Server
client.connect(function(err) {
  assert.equal(null, err);
  console.log("Connected successfully to server");

  const db = client.db(dbName);
  var runQuery = function(coll, query, iterations=1) {
      var start = Date.now();

      for (i = 0; i < iterations; i++) {
          res = db.collection(coll).find(query).toArray();
          }
      var end = Date.now();
      return end-start;
     };

   var a = runQuery("test.data", {"ident":"state-gcp-noregion-develop"},100);
   console.log(a);;
  client.close();
});

File: test_latency.py
import pprint
import time
import pymongo
from pymongo import MongoClient
from pymongo.read_preferences import Secondary
from pymongo import ReadPreference
client = pymongo.MongoClient("mongodb+srv://veronica:<PASSWORD>@pluto.e3h8i.mongodb.net/sample_mflix?retryWrites=true&w=majority")

col = client.test.data
start_time = time.time()
pprint.pprint(col.find_one({"ident":"state-gcp-noregion-develop"}))
print("--- %s seconds ---" % (time.time() - start_time))



client_us = pymongo.MongoClient('mongodb://veronica:<PASSWORD>@pluto-shard-00-01.e3h8i.mongodb.net:27017/sample_mflix?ssl=true&authSource=admin', read_preference=ReadPreference.SECONDARY)
db= client_us.test
col = db.data
start_time = time.time()
pprint.pprint(col.find_one({"ident":"state-gcp-noregion-develop"}))
print("US Secondary read --- %s seconds ---" % (time.time() - start_time))



client_sea = pymongo.MongoClient('mongodb://veronica:<PASSWORD>@pluto-shard-00-02.e3h8i.mongodb.net:27017/sample_mflix?ssl=true&authSource=admin', read_preference=ReadPreference.SECONDARY)
db = client_sea.test
col = db.data
start_time = time.time()
pprint.pprint(col.find_one({"ident":"state-gcp-noregion-develop"}))
print("SEA Secondary read --- %s seconds ---" % (time.time() - start_time))

